manim/animation/override_animation.py

```diff
@@ -0,0 +1,102 @@
+"""Decorator for Mobject-specific Animation overrides."""
+
+
+__all__ = ["override_animation", "_setup_animation_overriding"]
+
+from typing import Callable, Dict, List, Type
+
+from manim.mobject.mobject import Mobject
+from manim.utils.exceptions import MultiAnimationOverrideException
+
+from .. import logger
+
+
+def _setup_animation_overriding(
+    mobject_class: "Type(Mobject)",
+    overrides: "Dict[Type[Animation], List[Dict[Type[Mobject], str]]]",
```

Comment:
you don't need to use strings.
```suggestion
    mobject_class: Type(Mobject),
    overrides: Dict[Type[Animation], List[Dict[Type[Mobject], str]]],
```

Reply:
Also, why you should use `Type`?

Reply:
> Also, why you should use `Type`?

Type[X] means that a (sub)class of X is expected, not an instance.

Reply:
> you don't need to use strings.

As far as I know I do. Typing can be done with or without quotation marks - preferably without. However, that requires to import the class into the module, which in this case would cause circular import issues.

Reply:
> this case would cause circular import issues.

You should anyway import them even if it is a string for linters like mypy to know about them. To avoid circular import you can import it conditionally using https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING.

Reply:
...that creates a new problem: Now I get the error `NameError: name 'Animation' is not defined` when testing

Reply:
I changed it to only enquote Animation, not the whole type.

Reply:
As it states in the linked page: Only local variables can be typed using this method.

Reply:
> Now I get the error NameError: name 'Animation' is not defined when testing

You should import Animation conditionally and use it as a string. But since we have dropped support for py3.6 you can import `from __future__ import annotations` at the top of the file and then use then without quotes.

Reply:
Though you ignore it and use string and it's completely fine.

Reply:
ok ^^